README.md
DEATHGRIP
=========================
-------------------------


What is Deathgrip
-------------------------

Deathgrip is a Python-based program which is used to encrypt and decript plain text messages by converting letters to numbers and using a mathmatical algorithm based on the time, date, and random integers. 

 Basic Usage
-------------------------

Deathgrip is ran in either encoder (-e) or decoder (-d) mode. You must specify an input file. You can optionally specify an output file. To encode "foo.txt" you could use:

    deathgrip -e -i foo.txt

As it would by default send that output to "output.txt", you would decode the output with:

    deathgrip -e -i output.txt -o bar.txt

Since we specified an output file for the decoder, the decoded message can be seen by opening "bar.txt"
    
A complete list of command line arguments can be seen with 

	deathgrip --help
    

Instalation
--------------------------

The recommended debian based instalation method is as follows:

    sudo $SHELL
    cd /tmp/
    git clone https://github.com/super-nathan/deathgrip.git
    cd /tmp/deathgrip
    dpkg-buildpackage -us -uc
    cd ../
    dpkg -i deathgrip_*_all.deb
    exit
    
All other Linux users should
	
	sudo $SHELL
    cd /tmp/
    git clone https://github.com/super-nathan/deathgrip.git
    cd /tmp/deathgrip
    cp deathgrip /usr/bin/
    cp deathgrip.conf /etc/
    exit

Authors
------------------------

Joseph Ko and Nathan Weber


debian/changelog
deathgrip (0.2.1) unstable; urgency=low

  * added rudimentary manpage

 -- "Super" Nathan Weber <supernathansunshine@gmail.com>  Tue, 14 May 2013 16:01:16 +0900

deathgrip (0.2.0) unstable; urgency=low

  * Added support for configuration files

 -- "Super" Nathan Weber <supernathansunshine@gmail.com>  Tue, 14 May 2013 15:16:59 +0900

deathgrip (0.1.2) unstable; urgency=low

  * Making it more secure with a rando and the date

 -- "Super" Nathan Weber <supernathansunshine@gmail.com>  Wed, 17 Apr 2013 13:02:37 +0900

deathgrip (0.1.1) unstable; urgency=low

  * Added support for numbers and punctuation

 -- Joseph Ko <eaglclaws@hotmail.com>  Tue, 16 Apr 2013 12:52:49 +0900

deathgrip (0.1.0) unstable; urgency=low

  * Initial release. (Closes: #XXXXXX)

 -- "Super" Nathan Weber <supernathansunshine@gmail.com>  Mon, 15 Apr 2013 13:15:08 +0900


encode.py
#!/usr/bin/python3

## programming convention:
# I know "camel case" is a C codeing convention, but I like it!
# so please capitaliz the second word of a variable
# so Nathan's House as a variable would become: nathansHouse

## Transit string:
# as of now the tranmsited message (encodedMessage) is sent as follows:
# date,time,randomNumber, letter, letter, ................

#First we import some libraries. These will include additional functions that we need.
import math
import time
import random
import os
import sys
from optparse import OptionParser

# a quick definition
encodedMessage = ""
global fileName
dryRun = 0
outputFile = "output.txt"



# the parser is a better way to read command line arguments!
parser = OptionParser()
parser.add_option("-f", "--file", dest="filename", help="write report to FILE", metavar="FILE", action="store", type="string")
parser.add_option("-v", "--verbose", action="count", dest="verbosity", help="print output", default=0)  
parser.add_option("-d", "--dry-run", action="count", dest="dryrun", help="same as verbose with no output file", default=0)  
parser.add_option("-o", "--output", dest="outfile", help="specify output FILE", metavar="FILE", action="store", type="string", default="output.txt")  
(options, args) = parser.parse_args()
fileName = options.filename
verbosity = options.verbosity
outputFile = options.outfile
dryRun = options.dryrun
if dryRun > 0:
	verbosity = 1
if not options.filename: 
    parser.error("You must specify an input file")


# we are going to create a main function, this will be important later, when we need to decide if we are encoding or decoding
def main():
	makeTime();
	readFile();
	setupTransmit();
	doMath();
	writeFile();

	

def makeTime():
	global transmitTime
	global transmitDate
	global transmitRando
	global encodeDate
	global encodeTime
	global crazyMath
	global randDo
	randDo = random.randint(0,999)
	cleanTime = time.strftime('%H%M')
	cleanDate = time.strftime('%Y%m%d')
	transmitTime = int(cleanTime) + randDo
	transmitTime = int(transmitTime)**6
	transmitDate = int(cleanDate)**3
	transmitRando = int(randDo)**4
	crazyMath = cleanTime
	crazyMath += cleanDate



# I decided to use a random variable and add it to the time. The date will never repeat so there is
# no chance of a pattern being discovered by crackers. Hoewever, if a message was sent at the same 
# time everyday it would have the same time string. In order to confuse crackers we will also use
# a random number that we will transmit so that time is hidden
def setupTransmit():
	global transmitTime
	global transmitDate
	global encodedMessage
	transmitDateNew = str(transmitDate).zfill(30)
	transmitTimeNew = str(transmitTime).zfill(30)
	transmitRandoNew = str(transmitRando).zfill(30)
	encodedMessage += str(transmitDateNew)
	encodedMessage += str(transmitTimeNew)
	encodedMessage += str(transmitRandoNew)





def readFile():
	global workString
	openFile = sys.argv[1]
	workString = open(fileName, 'r').read()
	workString = workString.lower()
	if verbosity >= 1 :
		print (workString)

# this is the meat and potatos		
def doMath():
	global encodedMessage
	for char in workString:
			if char == "a":
				b = 12
			elif char == "b":
				b = 15
			elif char == "\n":
				b = 5
			elif char == "c":
				b = 17
			elif char == "d":
				b = 19
			elif char == "e":
				b = 23
			elif char == "f":
				b = 26
			elif char == "g":
				b = 28
			elif char == "h":
				b = 29
			elif char == "i":
				b = 37
			elif char == "j":
				b = 39
			elif char == "k":
				b = 40
			elif char == "l":
				b = 48
			elif char == "m":
				b = 51
			elif char == "n":
				b = 52
			elif char == "o":
				b = 53
			elif char == "p":
				b = 59
			elif char == "q":
				b = 70
			elif char == "r":
				b = 72
			elif char == "s":
				b = 77
			elif char == "t":
				b = 79
			elif char == "u":
				b = 80
			elif char == "v":
				b = 82
			elif char == "w":
				b = 86
			elif char == "x":
				b = 87
			elif char == "y":
				b = 90
			elif char == "z":
				b = 93
			elif char == " ":
				b = 99
			else:
				b = 0
			newLetter = int(b)*int(crazyMath)
			newLetter = newLetter**2
			newNewLetter = str(newLetter).zfill(30)
			encodedMessage += str(newNewLetter)
			

#Lets write this to an output file
def writeFile():
	if verbosity >= 1 :
		print(encodedMessage)
	if dryRun == 0 :
		f = open(outputFile, 'w')
		f.write(encodedMessage)
		f.close()

# Since EVERYTHING before this is definistions, running this one function, runs the program	
main();



old/decode.py
#!/usr/bin/python3

## programming convention:
# I know "camel case" is a C codeing convention, but I like it!
# so please capitaliz the second word of a variable
# so Nathan's House as a variable would become: nathansHouse

## Transit string:
# as of now the tranmsited message (encodedMessage) is sent as follows:
# date,time,randomNumber, letter, letter, ................

#First we import some libraries. These will include additional functions that we need.
import math
import time
import random
import os # we might not need this
import sys # we might not need this
from optparse import OptionParser

# a quick definition
encodedMessage = ""
global fileName
dryRun = 0
outputFile = "output.txt"



# the parser is a better way to read command line arguments!
parser = OptionParser()
parser.add_option("-f", "--file", dest="filename", help="write report to FILE", metavar="FILE", action="store", type="string")
parser.add_option("-v", "--verbose", action="count", dest="verbosity", help="print output", default=0)  
parser.add_option("-d", "--dry-run", action="count", dest="dryrun", help="same as verbose with no output file", default=0)  
parser.add_option("-o", "--output", dest="outfile", help="specify output FILE", metavar="FILE", action="store", type="string", default="output.txt")  
(options, args) = parser.parse_args()
fileName = options.filename
verbosity = options.verbosity
outputFile = options.outfile
dryRun = options.dryrun
if dryRun > 0:
	verbosity = 1
if not options.filename: 
    parser.error("You must specify an input file")


# we are going to create a main function, this will be important later, when we need to decide if we are encoding or decoding
def main():

	readMessage();
	receiveSalt();
	doMoreMath();
	writeDecodedFile();


def readMessage():
	global workStringTwo
	openFile = sys.argv[1]
	workStringTwo = open(fileName, 'r').read()
	if verbosity >= 1 :
		print (workStringTwo)


def receiveSalt():
	global receivedTime
	global receivedDate
	global receivedRando
	global decodedMessage
	global workStringTwo
	global crazyMathTwo
	decodedMessage = ""
	receivedDate = workStringTwo[:30]
	receivedDate = float(receivedDate)
	workStringTwo = workStringTwo[30:]
	receivedTime = workStringTwo[:30]
	receivedTime = float(receivedTime)
	workStringTwo = workStringTwo[30:]
	receivedRando = workStringTwo[:30]
	receivedRando = float(receivedRando)
	workStringTwo = workStringTwo[30:]
	realRando = math.pow(receivedRando, 1/4)
	realRando = float(realRando)
	realTime = math.pow(receivedTime, 1/6)
	realTime = realTime - realRando
	realTime = int(realTime)
	realDate = math.pow(receivedDate, 1/3)
	realDate = int(realDate)
	decodedMessage += "Time was:"
	decodedMessage += str(realTime)
	decodedMessage += "\n"	
	decodedMessage += "Date was:"
	decodedMessage += str(realDate)
	decodedMessage += "\n"	
	decodedMessage += "\n"
	realTime = str(realTime)
	realDate = str(realDate)
	crazyMathTwo = realTime
	crazyMathTwo += realDate
	crazyMathTwo = int(crazyMathTwo)


# this is the meat and potatos		
def doMoreMath():
	global workStringTwo
	global testVal
	global decodedMessage
	testVal = 0
	while len(workStringTwo) > 1:
		workStringThree = workStringTwo[:30]
		workStringThree = float(workStringThree)
		testVal = math.pow(workStringThree, 1/2)
		testVal = testVal / int(crazyMathTwo)
		testVal = int(testVal)
		if testVal == 12:
			b = "a"
		elif testVal == 15:
			b = "b"
		elif testVal == 5:
			b = "\n"
		elif testVal == 17:
			b = "c"
		elif testVal == 19:
			b = "d"
		elif testVal == 23:
			b = "e"
		elif testVal == 26:
			b = "f"
		elif testVal == 28:
			b = "g"
		elif testVal == 29:
			b = "h"
		elif testVal == 37:
			b = "i"
		elif testVal == 39:
			b = "j"
		elif testVal == 40:
			b = "k"
		elif testVal == 48:
			b = "l"
		elif testVal == 51:
			b = "m"
		elif testVal == 52:
			b = "n"
		elif testVal == 53:
			b = "o"
		elif testVal == 59:
			b = "p"
		elif testVal == 70:
			b = "q"
		elif testVal == 72:
			b = "r"
		elif testVal == 77:
			b = "s"
		elif testVal == 79:
			b = "t"
		elif testVal == 80:
			b = "u"
		elif testVal == 82:
			b = "v"
		elif testVal == 86:
			b = "w"
		elif testVal == 87:
			b = "x"
		elif testVal == 90:
			b = "y"
		elif testVal == 93:
			b = "z"
		elif testVal == 99:
			b = " "
		else:
			b = ""
		newLetter = str(b)
		decodedMessage += str(newLetter)
		workStringTwo = workStringTwo[30:]
			

#Lets write this to an output file
def writeDecodedFile():
	if verbosity >= 1 :
		print(decodedMessage)
	if dryRun == 0 :
		f = open(outputFile, 'w')
		f.write(decodedMessage)
		f.close()

# Since EVERYTHING before this is definistions, running this one function, runs the program	
main();

